src/api/Cosmos.ts

```diff
@@ -0,0 +1,385 @@
+
+    return data.withdraw_address;
+  } catch (e) {
+    return "";
+  }
+};
+
+export const getTransactions = async (address: string): Promise<any> => {
+  log("cosmjs", "fetch transactions");
+
+  try {
+    const perPage = 100;
+    const txs: { [id: string]: any } = {};
+    const data: Array<any> = [];
+    tmClient = await Tendermint34Client.connect(defaultRpcEndpoint);
+
+    // fetch incoming transactions
+    const txsIn = await tmClient.txSearch({
+      query: `transfer.recipient='${address}'`,
+      page: 1,
+      per_page: perPage,
+      order_by: "desc",
+    });
+
+    for (const tx of txsIn.txs) {
+      tx.hash = toHex(tx.hash).toUpperCase();
+      data.push(tx);
+    }
+
+    const txsInDone = txs.length;
+    const txsInNbPage = Math.ceil((txsIn.totalCount - txsInDone) / perPage);
+
+    // paginate over pages
+    for (let i = 2; i <= txsInNbPage; i++) {
+      const txsIn = await tmClient.txSearch({
+        query: `transfer.recipient='${address}'`,
+        page: i,
+        per_page: perPage,
+        order_by: "desc",
+      });
+
+      for (const tx of txsIn.txs) {
+        tx.hash = toHex(tx.hash).toUpperCase();
+        data.push(tx);
+      }
+    }
+
+    // fetch outgoing transactions
+    const txsOut = await tmClient.txSearch({
+      query: `message.sender='${address}'`,
+      page: 1,
+      per_page: perPage,
+      order_by: "desc",
+    });
+
+    for (const tx of txsOut.txs) {
+      tx.hash = toHex(tx.hash).toUpperCase();
+      data.push(tx);
+    }
+
+    const txsOutDone = txs.length;
+    const txsOutNbPage = Math.ceil((txsOut.totalCount - txsOutDone) / perPage);
+
+    // paginate over pages
+    for (let i = 2; i <= txsOutNbPage; i++) {
+      const txsOut = await tmClient.txSearch({
+        query: `message.sender='${address}'`,
+        page: i,
+        per_page: perPage,
+        order_by: "desc",
+      });
+
+      for (const tx of txsOut.txs) {
+        tx.hash = toHex(tx.hash).toUpperCase();
+        data.push(tx);
+      }
+    }
+
+    // fetch date and set fees
+    for (const tx of data) {
+      const block = await getBlock(tx.height);
+      tx.date = new Date(block.block.header.time);
+      tx.fee = new BigNumber(0);
+      const txRaw: DecodedTxRaw = decodeTxRaw(tx.tx);
+
+      if (txRaw.authInfo.fee) {
+        txRaw.authInfo.fee.amount.forEach((fee) => {
+          tx.fee = tx.fee.plus(fee.amount);
+        });
+      }
+    }
+
+    // sort transactions by date
+    data.sort((a, b) => {
+      return b.date.getTime() - a.date.getTime();
+    });
+
+    for (const tx of data) {
+      txs[tx.hash] = tx;
+    }
+
+    return txs;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+export const broadcast = async ({
+  signedOperation,
+}: {
+  signedOperation: SignedOperation;
+}): Promise<Operation> => {
+  log("cosmjs", "broadcast tx");
+
+  const { operation } = signedOperation;
+
+  const { data } = await network({
+    method: "POST",
+    url: `${defaultEndpoint}/txs`,
```

Comment:
As mentioned by e-mail, please try with the correct endpoint for new transactions format: `/cosmos/tx/v1beta1/txs`

Reply:
fix in 22a8559c